src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -148,25 +158,59 @@ public override IEnumerable<StringGetter[]> GetSignatures()
             return new[] { _args.Select<UDFArg, TexlStrings.StringGetter>(key => _ => key.NameIdent.Name.Value).ToArray() };
         }
 
+        public (IntermediateNode topNode, ScopeSymbol ruleScopeSymbol) GetIRTranslator()
+        {
+            return IRTranslator.Translate(_binding);
+        }
+
+        public RecordValue GetRuntimeRecordValue(FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            var argsArray = _args.ToArray();
+            var fieldNames = new List<NamedValue>();
+
+            for (int i = 0; i < argsArray.Length; i++)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var namedValue = new NamedValue(argsArray[i].NameIdent.Name.Value, args[i]);
+                fieldNames.Add(namedValue);
+            }
+
+            return FormulaValue.NewRecordFromFields(fieldNames);
+        }
+
         /// <summary>
         /// NameResolver that combines global named resolver and params for user defined function.
         /// </summary>
         private partial class UserDefinitionsNameResolver : INameResolver
         {
             private readonly INameResolver _globalNameResolver;
             private readonly IReadOnlyDictionary<string, UDFArg> _args;
+            private readonly FormulaValue[] _fvArgs;
```

Comment:
>FormulaValue [](http://example.com/codeflow?start=29&length=12)

Why not IEnumberable<>?

Reply:
might be nice to have count, indexing.  Maybe 
IReadOnlyCollection?

Reply:
Due to the new design this has been removed.